.travis.yml

```diff
@@ -75,13 +88,34 @@ script:
     - if [ "$FLAVOR" == "cs" ]; then xbuild /p:Configuration=Debug cs/cs.sln; fi
     - if [ "$FLAVOR" == "cs" ]; then xbuild /p:Configuration=Fields cs/cs.sln; fi
     - if [ "$FLAVOR" == "cs" ]; then mono NUnit.Runners.2.6.4/tools/nunit-console.exe -framework=mono-4.5 -labels cs/test/core/bin/debug/net45/Properties/Bond.UnitTest.dll cs/test/core/bin/debug/net45/Fields/Bond.UnitTest.dll cs/test/internal/bin/debug/net45/Bond.InternalTest.dll; fi
+
     - if [ "$FLAVOR" == "cpp-core" ]; then cmake -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE ..; fi
     - if [ "$FLAVOR" == "cpp-core" ]; then make --jobs 2 check; fi
+
     - if [ "$FLAVOR" == "cpp-comm" ]; then cmake -DBOND_ENABLE_COMM=TRUE -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE ..; fi
     - if [ "$FLAVOR" == "cpp-comm" ]; then make --jobs 2 check; fi
+
     - if [ "$FLAVOR" == "cpp-grpc" ]; then cmake -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DgRPC_ZLIB_PROVIDER=package ..; fi
     - if [ "$FLAVOR" == "cpp-grpc" ]; then make --jobs 2 check; fi
+
     - if [ "$FLAVOR" == "hs" ]; then cmake -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE ..; fi
     - if [ "$FLAVOR" == "hs" ]; then make gbc-tests; fi
     - if [ "$FLAVOR" == "hs" ]; then cd ../compiler; fi
     - if [ "$FLAVOR" == "hs" ]; then ../build/compiler/build/gbc-tests/gbc-tests; fi
+
+    - if [ "$FLAVOR" == "java" ]; then cmake -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE ..; fi
+    - if [ "$FLAVOR" == "java" ]; then make gbc; fi
+    - if [ "$FLAVOR" == "java" ]; then export PATH=`pwd`/compiler/build/gbc:$PATH; fi
+    - if [ "$FLAVOR" == "java" ]; then cd ../java/gradle-plugin; gradle build; fi
+    - if [ "$FLAVOR" == "java" ]; then mvn install:install-file -Dfile=build/libs/bond-gradle-1.0.jar -DgroupId=org.bondlib -DartifactId=gradle -Dversion=1.0 -Dpackaging=jar; fi
+    - if [ "$FLAVOR" == "java" ]; then cd ../core; gradle build; fi
+    - if [ "$FLAVOR" == "java" ]; then cd ../compat; gradle build; fi
```

Comment:
What happens if these gradle builds? We `set -e` below, which makes me think that the Travis build won't fail until much, much later.

Reply:
The build script will continue, but the build job will fail. This is how the other flavors work, though Java is obviously longer. This comment has unleashed a torrent of thoughts on shell and travis mechanics to justify those six characters.

AFAICT, the way travis works is that it pumps each of our lines into a single shell instance, wrapped inside a helper function. The wrapper provides the folding, pretty green/red text in the log, and implements `script_success ||= line_success` behavior. With flow control (`if [ "$FLAVOR"...` or semicolon-separated statements, the return code of the line is the return code of the last evaluated statement. (I wanted and had `cd core; gradle build; cd ..`, but then the line returns 0 even if gradle fails.)

The `for` loop to hit all examples was more annoying. Shell loops evaluate to the return code of the last statement in the last loop, so as long as the last example succeeds, the line with the loop succeeds from travis's perspective. `set -e` modifies the top-level shell so that as soon as any command, in any location, returns nonzero, the shell bails out with that retval. The effect is that the first failed example fails the build as you would expect. (Pedantic note: it also deprives the wrapper of its opportunity to do the nice folding and red/green stuff, but the gradle output from the failure will be the very last thing in the log.)

The right answer here is to have a process that programmatically discovers examples, runs them, and implements `examples_success ||= last_example_success`, which is what cmake will do once these examples are properly wired into it. I considered making a throwaway script, but this seemed like an easy and safe stopgap.

Reply:
Agreed that the build scripts could stand some refactoring. Not as part of this PR, though.